third_party/souffle/BUILD

```diff
@@ -93,7 +96,12 @@ cc_library(
 cc_binary(
     name = "souffle",
     srcs = ["src/main.cpp"],
-    copts = ["-std=c++17"],
+    copts = [
+        "-std=c++17",
+        # We don't care about non-critical issues in the third-party Souffle
```

Comment:
Just a nit: we do care as these warnings may be indicative of some underlying problem. However, I see your point about these wall of warnings obfuscating other warnings raised in our code.

Reply:
OK. Should I make the comment a little more nuanced in a follow-on PR? Something like "We don't want warnings in 3rd party packages to obscure issues in our own code that we are more likely to need to fix."?

Reply:
Not a big deal, but your suggestion sounds better.

Reply:
Softened the language as suggested.